internal/handler/msg_saslstart.go

```diff
@@ -129,3 +153,116 @@ func saslStartPlain(doc *types.Document) (string, string, error) {
 
 	return string(authcid), string(passwd), nil
 }
+
+// scramCredentialLookup looks up an user's credentials in the database.
+func (h *Handler) scramCredentialLookup(ctx context.Context, username, dbName string) (scram.StoredCredentials, error) {
```

Comment:
Why do we return `scram.StoredCredentials` and not `*scram.StoredCredentials` in this case? 
(Personally, I'm ok with both option but the option with pointer looks a bit more typical for the rest of the code, and also semantically it's easier to interpret (nil pointer is returner in case of an error)

Reply:
I didn't do that because I was trying to avoid this weird `return *cred, nil` line here, so as it wasn't necessary I left it as this, but I see it's happening just on a single place, so I'm changing it now.

```go
scramServer, err := scram.SHA256.NewServer(func(username string) (scram.StoredCredentials, error) {
		cred, err := h.scramCredentialLookup(ctx, username, dbName, "SCRAM-SHA-256")
		if err != nil {
			return scram.StoredCredentials{}, err
		}
		return *cred, nil
	})
	if err != nil {
		return "", err
	}
```

Reply:
FWIW, it is our documented best practice: https://github.com/FerretDB/FerretDB/blob/e1eaaad891265ac947fbf55fd5f736ba301c9bea/CONTRIBUTING.md?plain=1#L260-L261